worker/config/default.js

```diff
@@ -10,15 +10,16 @@ module.exports = {
 				wireless: process.env.NETWORK_WIRELESS_INTERFACE || 'wlan0',
 			},
 			qemu: {
-				network: {
-					bridgeName: process.env.QEMU_BRIDGE_NAME || 'br0',
-					bridgeAddress: process.env.QEMU_BRIDGE_ADDRESS || '192.168.100.1',
-					dhcpRange: process.env.QEMU_DHCP_RANGE || '192.168.100.2,192.168.100.254',
-				},
 				architecture: process.env.QEMU_ARCH || 'x86_64',
 				cpus: process.env.QEMU_CPUS || '4',
 				memory: process.env.QEMU_MEMORY || '2G',
 				debug: process.env.QEMU_DEBUG || false,
+				ports: {
+					vncMin: process.env.QEMU_VNC_PORT_MIN || 5900,
+					vncMax: process.env.QEMU_VNC_PORT_MAX || 6900,
+					qmpMin: process.env.QEMU_QMP_PORT_MIN || 5000,
+					qmpMax: process.env.QEMU_QMP_PORT_MAX || 5899,
+				}
```

Comment:
Don't kill me -- but now that I'm seeing this I think we might want to just expose `QEMU_VNC_PORT` and `QEMU_QMP_PORT` and use dynamic ports only if the value is not set?

That way we can manually specify ports per instance, or let QEMU figure it out with hardcoded ranges?

Sorry I didn't think of this sooner. I'm torn with wanting to be predictable but also scalable, and maybe the (better) compromise is be dynamic only when env vars are unset?

The same could be said for `QEMU_BRIDGE_ADDRESS` and `QEMU_DHCP_RANGE` and `QEMU_BRIDGE_NAME`? If not provided we will try to assign them dynamically with hardcoded ranges.

This feels much cleaner than exposing ranges. This way if you want to be 100% reproducible you can set every field used by QEMU.

Reply:
I agree, I'll make the change

Reply:
@klutchell take a look and see if this is ok - all parameters configurable through env vars - if the env var isn't set, then we dynamically generate them. I though I might as well make the port ranges configurable too in the case of the ports not being specified through the env vars/